src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -58,40 +59,71 @@ public void InitializeService(IProtocolEndpoint serviceHost)
         /// <returns></returns>
         internal async Task HandleRenameRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
         {
-            Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
-            ConnectionInfo connInfo;
+            Logger.Verbose("Handle Request in HandleRenameRequest()");
+            ExecuteActionOnObject(requestParams.ConnectionUri, requestParams.ObjectUrn, (dbObject) =>
+            {
+                var renamable = dbObject as IRenamable;
+                if (renamable != null)
+                {
+                    renamable.Rename(requestParams.NewName);
+                }
+                else
+                {
+                    throw new Exception(SR.ObjectNotRenamable(requestParams.ObjectUrn));
+                }
+            });
+            await requestContext.SendResult(true);
+        }
+
+        /// <summary>
+        /// Method to handle the delete object request
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute deletion operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal async Task HandleDropRequest(DropRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            Logger.Verbose("Handle Request in HandleDeleteRequest()");
+            ExecuteActionOnObject(requestParams.ConnectionUri, requestParams.ObjectUrn, (dbObject) =>
```

Comment:
should we use `DatabaseUtils.DoDropObject` to drop objects or is all the extra process that's doing unnecessary?  For context that code was ported directly from the "Drop Object" dialog in SSMS.

Reply:
for the object types we current support, what I have is good enough, but I ended up using what you suggested as it is more comprehensive.